swf/src/avm2/read.rs

```diff
@@ -171,6 +171,20 @@ impl<'a> Reader<'a> {
             0x1c => Multiname::MultinameLA {
                 namespace_set: self.read_index()?,
             },
+            0x1d => {
+                let base_type = self.read_index()?;
+                let count = self.read_u30()?;
+                let mut parameters = Vec::with_capacity(count as usize);
+
+                for _ in 0..count {
+                    parameters.push(self.read_index()?);
+                }
+
+                Multiname::TypeName {
+                    base_type,
+                    parameters,
+                }
+            }
```

Comment:
I personally feel like this would make more sense to error out on in `Multiname::from_abc_multiname`, rather than in `swf`. The latter is just a parsing library, and someone might want to be able to parse syntactically valid but semantically nonsensical bytecodes.

Plus, when we implement exception types, we're going to need to raise this as a `VerifyError`. AFAIK `swf` only returns boxed-string errors.

Reply:
@Herschel I implemented checks for multiple type parameters in `Multiname` - AFAIK that's the only thing we'd need to re-review before merging this.